include/drivers/pinctrl/stm32.h

```diff
@@ -0,0 +1,112 @@
+/*
+ * Copyright (c) 2020 Linaro Ltd.
+ * Copyright (c) 2021 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+/**
+ * @file
+ * STM32 SoC specific helpers for pinctrl driver
+ */
+
+#ifndef ZEPHYR_INCLUDE_DRIVERS_PINCTRL_PINCTRL_STM32_H_
+#define ZEPHYR_INCLUDE_DRIVERS_PINCTRL_PINCTRL_STM32_H_
+
+#include <devicetree.h>
+#include <zephyr/types.h>
+
+#ifdef CONFIG_SOC_SERIES_STM32F1X
+#include <dt-bindings/pinctrl/stm32f1-pinctrl.h>
+#else
+#include <dt-bindings/pinctrl/stm32-pinctrl.h>
+#endif
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+/** @cond INTERNAL_HIDDEN */
+
+/** Type for STM32 pin. */
+typedef struct pinctrl_pin {
+	/** Pinmux settings (port, pin and function). */
+	uint32_t pinmux;
+	/** Pin configuration (bias, drive and slew rate). */
+	uint32_t pincfg;
+} pinctrl_pin_t;
+
+/**
+ * @brief Utility macro to initialize pinmux field in #pinctrl_pin_t.
+ *
+ * @param node_id Node identifier.
+ */
+#define Z_PINCTRL_STM32_PINMUX_INIT(node_id) DT_PROP(node_id, pinmux)
+
+#ifdef CONFIG_SOC_SERIES_STM32F1X
+/**
+ * @brief Utility macro to initialize pincfg field in #pinctrl_pin_t (F1).
+ *
+ * @param node_id Node identifier.
+ */
+#define Z_PINCTRL_STM32_PINCFG_INIT(node_id)				       \
+	(((STM32_NO_PULL * DT_PROP(node_id, bias_disable))		       \
+	   << STM32_PUPD_SHIFT) |					       \
+	 ((STM32_PULL_UP * DT_PROP(node_id, bias_pull_up))		       \
```

Comment:
Nit: but I think you can stick these on one line so its:
`	 ((STM32_PULL_UP * DT_PROP(node_id, bias_pull_up)) << STM32_PUPD_SHIFT) | \`

Reply:
done